port_drayage_plugin/include/port_drayage_plugin/port_drayage_worker.h

```diff
@@ -109,22 +116,33 @@ namespace port_drayage_plugin
              * 
              * \param stop_speed_epsilon An epsilon factor to be used when
              * comparing the current vehicle's speed to 0.0
+             * 
+             * \param enable_port_drayage A boolean flag indicating whether port drayage
+             * operations are enabled or not. If false, the port drayage state machine
+             * will remain in an INACTIVE state.
+             * 
+             * \param call_set_active_route_client A lambda containing the logic
+             * necessary to call the SetActiveRoute service client. This lambda should
+             * contain all the necessary ROS logic so that it does not leak into 
+             * the implementation of this class.
              */
             PortDrayageWorker(
                 unsigned long cmv_id,
                 std::string cargo_id,
                 std::string host_id,
-                std::function<void(cav_msgs::MobilityOperation)> mobility_operations_publisher,
-                std::function<bool(cav_srvs::SetActiveRoute)> call_set_active_route_client, 
-                double stop_speed_epsilon) :
+                std::function<void(cav_msgs::MobilityOperation)> mobility_operations_publisher, 
+                double stop_speed_epsilon,
+                bool enable_port_drayage,
+                std::function<bool(cav_srvs::SetActiveRoute)> call_set_active_route_client) :
                 _cmv_id(cmv_id),
                 _cargo_id(cargo_id),
                 _host_id(host_id),
                 _publish_mobility_operation(mobility_operations_publisher),
                 _set_active_route(call_set_active_route_client),
-                _stop_speed_epsilon(stop_speed_epsilon) {
-                    initialize();
+                _stop_speed_epsilon(stop_speed_epsilon),
+                _enable_port_drayage(enable_port_drayage) {
                     _has_cargo = (_cargo_id == "") ? false : true;
```

Comment:
I have removed the `_has_cargo` element from PortDrayageWorker, since the information can already be inferred from `_cargo_id`. As a result, this `_has_cargo = (_cargo_id == "") ? false : true;` statement has been removed.